main.tf

```diff
@@ -29,6 +29,7 @@ locals {
     for k, v in module.spacelift_config.spacelift_stacks :
     k => v
     if
+    lookup(v.settings.spacelift, "admin_stack", null) != null ? v.settings.spacelift.admin_stack == data.spacelift_current_stack.administrative.id :
```

Comment:
Basically, check to see if we have an `admin_stack` setting for a specific stack.. if so, return `true` if the value matches the currently running stack. If it doesn't match, then return `false` so that the stack doesn't get included.

If no `admin_stack` is defined, then fallback to using the pre-existing `context_filter` logic.